schema/projectSchema.js

```diff
@@ -7,33 +7,70 @@ const projectSchema = object({
   description: string().required("Description is required").max(500, "Description must be under 500 characters"),
   eulogy: string().required("Eulogy is required").max(1000, "Eulogy must be under 1000 characters"),
-  causeOfDeath: string().required("Cause of death is required").max(255, "Keep cause of death under 255 characters"),
-  startDate: date().required("Start date is required").typeError("Start date must be a valid date"),
-  endDate: date()
-    .required("End date is required")
-    .typeError("End date must be a valid date")
-    .when("startDate", (startDate, schema) =>
-      startDate ? schema.min(startDate, "End date must be after start date") : schema
-    ),
-  types: array().of(number()).min(1, "At least one type must be selected").required("Types are required"),
-  tombstoneId: number().nullable(),
-});
+  causeOfDeath: string().required("Cause of death is required").max(100, "Keep cause of death under 100 characters"),
+  startDate: date()
+    .required("Start date is required")
+    .typeError("Start date must be a valid date")
+    .transform((v, o) => (o === "" || o === undefined ? null : v)),
 
-const projectUpdateSchema = object({
-  name: string().min(1, "Name can't be empty").max(100, "Name must be under 100 characters"),
-  description: string().max(500, "Description must be under 500 characters"),
-  eulogy: string().max(1000, "Eulogy must be under 1000 characters"),
-  causeOfDeath: string().max(255, "Keep cause of death under 255 characters"),
-  startDate: date().typeError("Start date must be a valid date"),
   endDate: date()
+    .required("End date is required")
     .typeError("End date must be a valid date")
-    .when("startDate", (startDate, schema) =>
-      startDate ? schema.min(startDate, "End date must be after start date") : schema
-    ),
-  tombstoneId: number().nullable(),
-  status: string().oneOf(["inactive", "active", "buried", "resurrected", "completed", "archived"]),
+    .transform((v, o) => (o === "" || o === undefined ? null : v))
+    .when("startDate", (start, schema) => {
+      const startDate = new Date(start);
+      return !isNaN(startDate) ? schema.min(startDate, "End date must be after start date") : schema;
+    }),
+  types: array()
+    .of(number())
+    .min(1, "At least one type must be selected")
+    .test("unique", "Cannot contain duplicate types.", (value) =>
+      value ? value.length === new Set(value)?.size : true
+    )
+    .required("Types are required"),
+  tombstoneId: number("Tombstone ID must be a number").typeError("Tombstone ID must be a number").nullable(),
 });
 
+const projectUpdateSchema = object()
+  .shape({
+    name: string().min(1, "Name can't be empty").max(100, "Name must be under 100 characters"),
+    description: string().max(500, "Description must be under 500 characters"),
+    eulogy: string().max(1000, "Eulogy must be under 1000 characters"),
+    causeOfDeath: string().max(100, "Keep cause of death under 255 characters"),
```

Comment:
Make sure it matches with createProject, and decide on a 100 or 255 characters
```suggestion
    causeOfDeath: string().max(255, "Keep cause of death under 255 characters"),
```
or:

```suggestion
    causeOfDeath: string().max(100, "Keep cause of death under 100 characters"),
```